packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -216,20 +238,28 @@ type CanvasInfo = {
   height: number;
   canvasWidth: number;
   canvasHeight: number;
+  proportion: number;
   viewport: PdfjsLib.PDFPageViewport;
 };
 
 function getCanvasInfo(
   loadedPage: PdfjsLib.PDFPageProxy | null | undefined,
-  scale: number
+  scale: number,
+  rootRef: RefObject<HTMLDivElement>
 ): CanvasInfo | null {
-  const canvasScale = window.devicePixelRatio ?? 1;
-  const viewport = loadedPage?.getViewport({ scale: scale * canvasScale });
-  if (viewport) {
+  const rootDimensions = rootRef?.current?.getBoundingClientRect();
+  if (loadedPage && rootDimensions) {
+    const canvasScale = window.devicePixelRatio ?? 1;
+    const width = rootDimensions.width * scale;
+    const pageWidth = loadedPage.view[2];
```

Comment:
What is `view[2]`?

Reply:
Good question, and I'm glad you asked. I thought it was the inherent width of the PDF, but it's actually the `x2` of the visible portion of the PDF (https://mozilla.github.io/pdf.js/api/draft/module-pdfjsLib-PDFPageProxy.html#view). So we'll need to adjust this to `loadedPage.view[2] - loadedPage.view[0]`, and make sure there's no unit conversion.

The reason that this is necessary is that we need to determine the ratio of available width to the real width of the PDF in order to incorporate that value into the scale being sent into `getViewport`.

Reply:
This concerns me. "Unit conversion", "is often", etc. Is there no function that returns what you need in a predictable manner?

Reply:
We could create one viewport with a scale value of `scale * canvasScale` and use its width / height to generate the `proportion` value necessary to create a second viewport with scale value of `scale * canvasScale * proportion`, but that seemed convoluted.

As far as I can tell, the unit conversion is what we're already doing with the device pixel ratio. `loadedPage.view` will often be `[0, 0, width, height]`, since the PDF will usually start at (0,0) and then continue to (width, height). But occasionally, it would go from (x1, y1) to (x1 + width, y1 + height) `loadedPage.view = [x1, y1, x1 + width, y1 + height]`. Setting `pageWidth = loadedPage.view[2] - loadedPage.view[0]` should be correct in either case, even if we're often hitting the first case (in which `pageWidth = loadedPage.view[2]` would be sufficient.

Reply:
To be clear, we need the `proportion` value prior to the creation of the real viewport to prevent the blurriness that you were seeing previously.